src/platform/log.c

```diff
@@ -29,3 +42,19 @@ void log_error(const char *msg, const char *param_str, int param_int)
 {
     SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "%s", build_message(msg, param_str, param_int));
 }
+
+void log_window_init() {
```

Comment:
`log_window_init` and `log_window_draw` should be in a new file called `src/window/log.c`. This is because of code structure, organization and clarity - drawing functions should be in the `src/window` subfolder.

In `log_window_draw` you could have a `const uint8_t  *log_fetch_line(int line)`, which is defined in `platform/log.c` and fetches the line in order (it returns `log_history_buf[(i + log_history_index)%40]`).